collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormDetails.java

```diff
@@ -28,22 +28,22 @@
     private final String formID;
     private final String formVersion;
     private final String hash;
-    private final String manifestFileHash;
     private final boolean isNotOnDevice;
     private final boolean isUpdated;
+    private final ManifestFile manifest;
 
     public ServerFormDetails(String formName, String downloadUrl, String manifestUrl, String formID,
-                             String formVersion, String hash, String manifestFileHash,
-                             boolean isNotOnDevice, boolean isUpdated) {
+                             String formVersion, String hash,
+                             boolean isNotOnDevice, boolean isUpdated, ManifestFile manifest) {
```

Comment:
We now just always include the manifest as part of the form details so logic that deals with it can be simpler (and doesn't end up fetching the manifest when it doesn't need to).